containers/ecr-viewer/src/app/api/services/core_database_repo.ts

```diff
@@ -0,0 +1,337 @@
+import { Kysely } from "kysely";
+
+import {
+  ECRConditions,
+  NewECRConditions,
+  ECRConditionsUpdate,
+  ECRRuleSummaries,
+  NewECRRuleSummaries,
+  ECRRuleSummariesUpdate,
+  ECR,
+  NewECR,
+  ECRUpdate,
+} from "./core_types";
+import { Core } from "./core_types";
+import { db } from "./database";
+
+// ECR_DATA
+
+/**
+ * Finds an eICR by its ID
+ * @async
+ * @function findEcrById
+ * @param id - thd ID of the eCR to be looked up
+ * @returns an eICR object
+ */
+export async function findEcrById(id: string | null): Promise<ECR | undefined> {
+  if (!id) {
+    throw new Error("eICR ID is required.");
+  }
+  try {
+    return await (db as Kysely<Core>)
+      .selectFrom("ecr_data")
+      .where("eICR_ID", "=", id)
+      .selectAll()
+      .executeTakeFirst();
+  } catch (error) {
+    console.error(error);
+  }
+}
+
+/**
+ * Finds an eICR by its criteria
+ * @async
+ * @function findEcr
+ * @param criteria - the Partial<ECR> used to filter against
+ * @returns an eICR object
+ */
+export async function findEcr(criteria: Partial<ECR> | null): Promise<ECR[]> {
+  let query = (db as Kysely<Core>).selectFrom("ecr_data");
+
+  if (!criteria || criteria === null) {
+    throw new Error("eICR Criteria is required.");
+  }
+
+  for (const criterium of Object.keys(criteria) as (keyof ECR)[]) {
+    if (criteria[criterium] !== undefined && criteria[criterium] !== null) {
+      query = query.where(criterium, "=", criteria[criterium]);
```

Comment:
do we need to handle the `null` case? in a prior iteration that was searched for for one of them